extensions/wikia/AdEngine/js/video/uapVideo.js

```diff
@@ -2,96 +2,135 @@
-			remainingTime = ima.adsManager.getRemainingTime();
-
-		if (remainingTime) {
-			currentTime.style.transitionDuration = remainingTime + 's';
-			currentTime.style.width = '100%';
-		} else {
-			currentTime.style.width = '0';
-		}
+	function loadPorvata(params, slotContainer, providerContainer) {
+		params.container = slotContainer;
+
+		log(['VUAP loadPorvata', params], log.levels.debug, logGroup);
+
+		return porvata.inject(params)
+			.then(function (video) {
+				videoInterface.setup(video, [
+					'progressBar',
+					'pauseOverlay',
+					'volumeControl',
+					'closeButton',
+					'toggleAnimation'
+				], {
+					image: providerContainer,
+					container: slotContainer,
+					aspectRatio: params.aspectRatio,
+					videoAspectRatio: params.videoAspectRatio
+				});
+
+				video.addEventListener('allAdsCompleted', function () {
+					video.reload();
+				});
+
+				return video;
+			});
 	}
 
-	function addProgressListeners(video) {
-		video.addEventListener('start', updateProgressBar);
-		video.addEventListener('resume', updateProgressBar);
-		video.addEventListener('complete', updateProgressBar);
-		video.addEventListener('pause', function (ima) {
-			var progressBar = ima.container.querySelector('.progress-bar'),
-				currentTime = progressBar.querySelector('.current-time');
+	function loadPlaywire(params, adSlot, imageContainer) {
+		var container = doc.createElement('div');
+
+		container.classList.add('video-player', 'hidden');
+		adSlot.appendChild(container);
+
+		params.container = container;
+
+		log(['VUAP loadPlaywire', params], log.levels.debug, logGroup);
+		return playwire.inject(params)
+			.then(function (video) {
+				videoInterface.setup(video, [
+					'closeButton',
+					'toggleAnimation'
+				], {
+					image: imageContainer,
+					container: adSlot,
+					aspectRatio: params.aspectRatio,
+					videoAspectRatio: params.videoAspectRatio
+				});
+
+				video.addEventListener('wikiaAdStarted', function () {
+					var slotWidth = getSlotWidth(adSlot);
+					video.resize(slotWidth, getVideoHeight(params));
+				});
+				if (params.autoplay) {
+					var slotWidth = getSlotWidth(adSlot);
+					video.play(slotWidth, getVideoHeight(params));
+				}
 
-			currentTime.style.width = (currentTime.offsetWidth / progressBar.offsetWidth * 100) + '%';
-		});
+				return video;
+			});
 	}
 
-	function loadVideoAd(params, adSlot, imageContainer, getSlotWidth) {
-		getSlotWidth = getSlotWidth || defaultGetSlotWidth;
-
-		try {
-			var video = videoAdFactory.create(
-				getSlotWidth(adSlot),
-				getVideoHeight(getSlotWidth(adSlot), params),
-				adSlot,
-				{
-					src: params.src,
-					pos: params.slotName,
-					uap: uapContext.getUapId(),
-					passback: 'vuap'
-				}
-			);
-			addProgressListeners(video);
+	function loadVideoAd(params) {
+		var loadedPlayer,
+			providerContainer = adSlot.getProviderContainer(params.slotName),
+			slotContainer = providerContainer.parentNode,
+			videoWidth = getSlotWidth(slotContainer);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the declaration of the unused 'videoWidth' variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "videoWidth" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)